### ISSUE TITLE
Support for Rustdesk

### ISSUE THREAD

**Issue by Wargamer-Senpai** [xpipe-io/xpipe#384]
Would it be possible to integrate support for Rustdesk, to call it, for example (Windows): 
```powershell
C:\Program Files\RustDesk\rustdesk.exe --connect <id> --password <password>
```

**Comment by crschnick** [comment]
Yes, if that is the only thing to add. I have not used rustdesk yet. But if there are other authentication measures, I'm not sure how well these could be supported

**Comment by Wargamer-Senpai** [comment]
Would be cool. I've looked into the [documentation](https://rustdesk.com/docs/en/client/#command-line-parameters), they have referenced this [file](https://github.com/rustdesk/rustdesk/blob/bdc5cded221af9697eb29aa30babce75e987fcc9/src/core_main.rs#L456) where all their parameters are listed/documented and it looks like this is the only method of authentication.

**Comment by crschnick** [comment]
And they don't have their own graphical launcher? XPipe wouldn't be more than just that

**Comment by Wargamer-Senpai** [comment]
They do have a graphical launcher, but it would be nicer if it would be all in one place and as a bonus have it on all devices synced 😉

**Comment by nativeit** [comment]
Just to add what (relatively little) I know that might be relevant. RustDesk can be called via a custom URL scheme, `rustdesk://connection/new/${rustdesk_id}?password=${rustdesk_pwd}` so one option would be to simply add a command/script to the host that launches that URL, and if you'd care to setup additional functions to automatically fill the ID and password, there are ways to achieve that using whichever scripting language you're comfortable with and/or makes sense for the platform.

For Windows machines, there are some PowerShell scripts that I have personally tested (and currently use) that automate installing RustDesk (with a custom ID/relay/API server), as well as for retrieving the ID number and password, in the documentation for integrating RustDesk with Tactical RMM ([link](https://docs.tacticalrmm.com/3rdparty_rustdesk/)). You would need to use WinRM/Remote Powershell, but happily XPipe seems pretty well suited for that!

So even if there's not enough interest to warrant including the RustDesk services/protocols in XPipe natively, it's script-able enough to establish some templates that would provide a similar experience. I use RustDesk on a number of machines, I may end up writing those scripts myself, and if so I will gladly publish them here for anyone else who may want them. In the meantime, I'd love to know if anyone else does anything with this.

**Comment by crschnick** [comment]
Does rustdesk store previous connections on the system somewhere that xpipe can detect automatically?

**Comment by Wargamer-Senpai** [comment]
Looks like `C:\Users\%username%\AppData\Roaming\RustDesk\config\peers`, the filename is the id and inside the files are more relevant information (hostname, password, connection options....)
